### ISSUE TITLE
Dynamic learning parameter - time period

### ISSUE THREAD

**Issue by jSaso** [deepjavalibrary/djl#37]
## Description
Dynamically modified learning parameter over time. We want learning parameter to change dynamically when network is learning:
- constant learning parameter at the start then **cycle** mode. We want to start with constant learning parameter, then after N epoch start **cycle** - like saw, from min. to max. definition of learning parameter (up and down)
- it should be configurable - we want to start with static learning parameter and then **cycle**, we want to start with **cycle** at the beginning,.. 
- any combination with constant and **cycle**, for example 0-10 epoch constant learning parameter, then **cycle**(whatever type) (11-20), then again constant parameter (21-30),..

**Cycle** (learning parameter alternation) mode should have multiple types, with configurable time interval and min., max. learning parameter value, for example:
- like saw wave (up and down /\ /\ /\ )
- like first half of saw wave (just up / / / )
- like second half of saw wave (just down \ \ \ )
- we can go any combination (/\ - / / - \ \ )

### Will this change the current api? How?
Yes, implement learning parameter which dynamically changes over iteration/epoch

### Who will benefit from this feature?
Everybody, this will enable network to learn more quickly.

## References
Here is dynamic learning parameter change implementation, kind of saw wave over time:
https://github.com/eclipse/deeplearning4j/blob/b5f0ec072f3fd0da566e32f82c0e43ca36553f39/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/schedule/CycleSchedule.java
We want to extend this implementation - see above.

**Comment by zachgk** [comment]
The cycle learning rate(s) can be added by extending the [LearningRateTracker](https://github.com/awslabs/djl/blob/master/api/src/main/java/ai/djl/training/optimizer/learningrate/LearningRateTracker.java), either for DJL or your own work. This would also be a good first issue for community contributors.
